src/utils.py

```diff
@@ -1,67 +1,86 @@
+def close_connection(conn, cursor):
+    cursor.close()
+    conn.close()
 
-    with open(schema_path) as create_file:
+def create_database_table(conn, cursor, table_name, database, sql_create):
+    with open(sql_create) as create_file:
         create_query = "".join(create_file.readlines())
         cursor.execute(create_query)
         conn.commit()
-    cursor.close()
-    conn.close()
-
-def delete_existing_records(schema_name, database):
-    conn = connect(database)
-    cursor = conn.cursor()
+        print("{} table successfully created in {} database.".format(table_name, database))
 
-    delete_query = 'TRUNCATE TABLE {}'.format(schema_name)
+def delete_table_records(conn, cursor, table_name):
+    delete_query = 'TRUNCATE TABLE {}'.format(table_name)
     cursor.execute(delete_query)
     conn.commit()
+    print("Successfully truncated table {}.".format(table_name))
 
-    cursor.close()
-    conn.close()
-
-def extract_data(file_path, flag, schema_name, database):
-    conn = connect(database)
-    cursor = conn.cursor()
-
-    if flag:
-        delete_existing_records(schema_name, database)
-    
-    if schema_name == 'raw_timesheet_data':
-        insert_path = '../sql/query/insert_raw_timesheet_data.sql'
-    elif schema_name == 'raw_employee_data':
-        insert_path = '../sql/query/insert_raw_employee_data.sql'
-
+def extract_data_from_csv(conn, cursor, file_path, table_name, sql_insert):
     with open(file_path, 'r') as file:       
         next(file)
         for line in file:
             row = line.strip().split(',')
-
-            with open(insert_path) as insert_file:
+            with open(sql_insert) as insert_file:
                 insert_query = "".join(insert_file.readlines())
                 cursor.execute(insert_query, row)
                 conn.commit()
-    cursor.close()
-    conn.close()
\ No newline at end of file
+        print("Values successfully inserted in {} table.".format(table_name))
+
+def extract_data_from_file(conn, cursor, file_path, table_name, sql_insert):
+    file_type = file_path.split('.')[-1] # json, csv)
+
+    if file_type == 'csv':       
+        extract_data_from_csv(conn, cursor, file_path, table_name, sql_insert)
+
+    elif file_type == 'json' or file_type == 'xml':
+        xml_flag = False
+        with open(file_path) as file_data:
+            if file_type == 'json':
+                record_list = json.load(file_data)
+            elif file_type == 'xml':
+                data_dict = xmltodict.parse(file_data.read())
+                xml_flag = True
+            
+            if xml_flag: 
+                record_list = data_dict["EmployeeList"]["Employee"]
+        
+        with open(sql_insert) as insert_file:
+            insert_query = "".join(insert_file.readlines())
+            cursor.executemany(insert_query, record_list)
+            conn.commit()
+
+def extract_data_from_db(source_cursor, dest_conn, dest_cursor, source_db, dest_db, sql_select, sql_insert):
```

Comment:
Will source db and destination db always be in same database server?

Reply:
Yesko laagi chahi didi we can connect to other db as well. We can define the source and destination dbs in the extract_from_dbs bhanne list of dictionaries.